### PR TITLE
[1712] Fix an issue that triggers direct label edition even if there …

### LINKED ISSUE #1712: F2 and double-click triggers direct edition of edge labels even when there is no corresponding tool
When an edge does not provide a (center) label edit tool, the corresponding tool does not appear in the edge's palette.
However it is still possible to trigger the first part of the direct edit workflow by hitting F2, double clicking on the edge or starting to type a new label.

### PR DESCRIPTION
**Pull request by frouene**
…is no corresponding tool

Bug: https://github.com/eclipse-sirius/sirius-web/issues/1712

# Pull request template

## General purpose
What is the main goal of this pull request?
- [x] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -25,6 +25,7 @@ UI is unchanged except that we removed the transactional behavior of the _Transf
 - https://github.com/eclipse-sirius/sirius-web/issues/2269[#2269] [core] The GraphQL field `EditingContext#domains` has been modified.
 A new parameter `rootDomainsOnly: Boolean` has been added, allowing to remove domains not compatible with root objects.
 For example, Diagram & Form domains are not root domains because `DiagramDescription` and `FormDescription` are not root objects valid candidates.
+- https://github.com/eclipse-sirius/sirius-web/issues/1712[#1712] [diagram] The GraphQL type `Node` must now implement `isLabelEditable` boolean field.
```
Comment (pcdavid) [inline CHANGELOG.adoc:28]: The original issue (https://github.com/eclipse-sirius/sirius-web/issues/1712) was about edges, not nodes.

File: packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/NodeComponent.java [hunk 2]
```diff
@@ -283,17 +284,14 @@ private CollapsingState computeCollapsingState(String nodeId, Optional<Node> opt
...
-     *            The optional diagram event modifying the default modifier set of the node
-     * @param optionalPreviousNode
-     *            The previous node from which get the old modifier set. If empty, the old modifier set is set to an
-     *            empty Set
-     * @param id
-     *            The ID of the current node
+     * @param optionalDiagramEvent The optional diagram event modifying the default modifier set of the node
+     * @param optionalPreviousNode The previous node from which get the old modifier set. If empty, the old modifier set is set to an
+     *                             empty Set
+     * @param id                   The ID of the current node
```
Comment (pcdavid) [inline packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/NodeComponent.java]: Unrelated formatting change (inconsistent with what Eclipse produces).

File: packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/NodeComponent.java [hunk 3]
```diff
@@ -387,7 +382,8 @@ private List<Element> getBorderNodes(Optional<Node> optionalPreviousNode, Variab
                 .forEach(borderNodeDescriptions::add);
 
         return borderNodeDescriptions.stream().map(borderNodeDescription -> {
-            List<Node> previousBorderNodes = optionalPreviousNode.map(previousNode -> new DiagramElementRequestor().getBorderNodes(previousNode, borderNodeDescription)).orElse(List.of());
+            List<Node> previousBorderNodes = optionalPreviousNode.map(previousNode -> new DiagramElementRequestor().getBorderNodes(previousNode, borderNodeDescription))
+                    .orElse(List.of());
```
Comment (pcdavid) [inline packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/NodeComponent.java]: Unrelated formatting change (inconsistent with what Eclipse produces).

File: packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/NodeComponent.java [hunk 4]
```diff
@@ -421,7 +417,8 @@ private List<Element> getChildNodes(Optional<Node> optionalPreviousNode, Variabl
                 .forEach(childNodeDescriptions::add);
 
         return childNodeDescriptions.stream().map(childNodeDescription -> {
-            List<Node> previousChildNodes = optionalPreviousNode.map(previousNode -> new DiagramElementRequestor().getChildNodes(previousNode, childNodeDescription)).orElse(List.of());
+            List<Node> previousChildNodes = optionalPreviousNode.map(previousNode -> new DiagramElementRequestor().getChildNodes(previousNode, childNodeDescription))
```
Comment (pcdavid) [inline packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/NodeComponent.java:425]: Unrelated formatting change (inconsistent with what Eclipse produces).

File: packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/NodeComponent.java [hunk 5]
```diff
@@ -341,12 +339,9 @@ private ViewModifier computeState(Set<ViewModifier> modifiers) {
...
      *
-     * @param optionalPreviousNode
-     *            The previous node if this node existed during a previous rendering
-     * @param nodeDescription
-     *            The description of the node
-     * @param nodeVariableManager
-     *            The variable manager of the node
+     * @param optionalPreviousNode The previous node if this node existed during a previous rendering
+     * @param nodeDescription      The description of the node
+     * @param nodeVariableManager  The variable manager of the node
```
Comment (pcdavid) [inline packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/NodeComponent.java]: Unrelated formatting change (inconsistent with what Eclipse produces).

**Review by pcdavid — APPROVED** [review]
There are some unrelated formatting changes, but nothing worth blocking the merge.

File: CHANGELOG.adoc (hunk 1 above)
Comment (frouene) [reply CHANGELOG.adoc:28]: The direct edit is only possible from the palette tool for the edge with react-flow.
The same behavior should be implemented when we add the shortcuts

**Review by pcdavid — APPROVED** [review]
